packages/plugin-ext/src/hosted/node/plugin-service.ts

```diff
@@ -82,22 +82,25 @@ export class HostedPluginServerImpl implements HostedPluginServer {
         return [...plugins.values()];
     }
 
-    async getDeployedPlugins({ pluginIds }: GetDeployedPluginsParams): Promise<PluginMetadata[]> {
+    async getDeployedPlugins({ pluginIds }: GetDeployedPluginsParams): Promise<DeployedPlugin[]> {
         if (!pluginIds.length) {
             return [];
         }
         const plugins = [];
         let extraPluginMetadata: Map<string, PluginMetadata> | undefined;
         for (const pluginId of pluginIds) {
-            let plugin = this.deployerHandler.getDeployedPluginMetadata(pluginId);
+            let plugin = this.deployerHandler.getDeployedPlugin(pluginId);
             if (!plugin) {
                 if (!extraPluginMetadata) {
                     extraPluginMetadata = new Map<string, PluginMetadata>();
-                    for (const metadata of await this.hostedPlugin.getExtraPluginMetadata()) {
-                        extraPluginMetadata.set(metadata.model.id, metadata);
+                    for (const extraMetadata of await this.hostedPlugin.getExtraPluginMetadata()) {
+                        extraPluginMetadata.set(extraMetadata.model.id, extraMetadata);
                     }
                 }
-                plugin = extraPluginMetadata.get(pluginId);
+                const metadata = extraPluginMetadata.get(pluginId);
+                if (metadata) {
+                    plugin = { metadata };
```

Comment:
@benoitf not sure whether extra plugin metadata suppose to have any contributes

It would be good to setup the example app using these APIs or remove and move them to Che, no one except Che developers can reason about them right now. cc @svenefftinge

Reply:
@akosyakov until now examples are not examples but just assemblies of theia extensions from built-in `packages`

Reply:
> not sure whether extra plugin metadata suppose to have any contributes 

What about apis? Should extra plugin metadata be changed to provide something else as well?

Reply:
> until now examples are not examples but just assemblies of theia extensions from built-in packages

My point is that there should be a criteria to decide what should be a contribution point or what should be just overloaded via DI in a concrete product. Till now, in the rest of Theia, the criteria was to have at least one 2-3 stable internal clients to the contribution point.  It was done to make sure that:
- anyone can judge about usefulness of such APIs, not guess or wait for someone
- don't have complexity of individual solutions in the framework, gain experience before abstracting something
- don't have unused contribution points

In the plugin system I see information passed around and contribution points which are never used. For a Theia committer who is not aware of Che, it looks confusing and unnecessary. Why should plugin host have an ID of an individual plugin when it runs many? Why do we have host always when we have one? What is extra metadata? and so on.

>  until now examples are not examples but just assemblies of theia extensions from built-in packages

We always treated it as an example. Although it should be aligned with there we are going, i.e. use VS Code extension as you mentioned. If the plugin system has support of remote running, adding some simple example would help with reasoning about unused data and contribution points as well.